prow/cmd/deck/main.go

```diff
@@ -105,10 +109,27 @@ func main() {
 			path: *tideURL,
 		}
 		ta.start()
-		http.Handle("/tide.js", gziphandler.GzipHandler(handleTide(configAgent, ta)))
+		mux.Handle("/tide.js", gziphandler.GzipHandler(handleTide(configAgent, ta)))
 	}
 
-	logger.WithError(http.ListenAndServe(":8080", nil)).Fatal("ListenAndServe returned.")
+	// optionally inject http->https redirect handler when behind loadbalancer
+	if *redirectHTTPTo != "" {
+		redirectMux := http.NewServeMux()
+		redirectMux.Handle("/", func(oldMux *http.ServeMux, host string) http.HandlerFunc {
+			return func(w http.ResponseWriter, r *http.Request) {
+				if r.Header.Get("x-forwarded-proto") == "http" {
+					redirectURL, _ := url.Parse(r.URL.String())
+					redirectURL.Scheme = "https"
+					redirectURL.Host = host
```

Comment:
The `host` parameter isn't needed. `*redirectHTTPTo` can be used directly here.

Reply:
It can, but it seemed cleaner to also capture this, and we really don't need to pointer deref every time because this can't change (and I'm doubtful that go would attempt to optimize this)